### PR TITLE
Reset data layer when tracking a new page

### PR DESCRIPTION
**Pull request by dpobel**
That way when changing page, we start from a fresh state

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -27,6 +27,16 @@ var GTM = (module.exports = integration("Google Tag Manager")
     '<script src="//www.googletagmanager.com/gtm.js?id={{ containerId }}&l=dataLayer&gtm_preview={{ environment }}">'
   ));
 
+const buildResetObjects = (props) => {
+  const resetPreviousProps = {};
+  Object.keys(props).forEach((key) => {
+    if (key !== "gtm" && key !== "userConsents") {
+      resetPreviousProps[key] = null;
```
Comment (real34) [inline lib/index.js]: How are data appearing in the datalayer then? Still there, or not visible at all?

I remember seeing an advice recommending to use `undefined` instead:
```suggestion
      resetPreviousProps[key] = undefined;
```

If you tried it and it works, don't worry with this!

**Review by real34 — APPROVED** [review]
It seems good indeed. Thanks!

File: lib/index.js (hunk 1 above)
Comment (dpobel) [reply lib/index.js]: For instance in a scenario where, I first view the homepage, then navigates to a category and then to the cart, the data layer as seen in GTM debugger has the following data:

```js
{
  event: "Cart Viewed",
  gtm: {uniqueEventId: 11, start: 1634120474595},
  userConsents: ["google-tag-manager"],
  name: "Cart",
  path: "/cart",
  referrer: "",
  search: "",
  title: "My Cart",
  url: "http://front-commerce.localhost:4000/cart",
  segmentAnonymousId: "bff39518-55a1-4d85-9b26-14f6948634d3",
  list_id: null,
  category: null,
  products: [
    {
      sku: "MS04-M-Orange",
      name: "Gobi HeatTec&reg; Tee",
      quantity: 1,
      price: 29,
      position: 1
    }
  ],
  cart_id: "226"
}
```

as you can see `list_id` which comes from the category page is still there but with `null`. If I apply you suggestion, it is also there but with `undefined`

Comment (real34) [reply lib/index.js]: Thanks!
That should indeed be good then.

**Pull request merged**